### DISCUSSION TITLE
How to keep two hard drives in sync and set the HDD that was cloned as the new backup target?

### DISCUSSION THREAD

**Discussion by mYnDstrEAm** [borgbase/vorta#1730] [FAQ]
I have a hard drive and clone it to a new hard drive. Because the former hard drive is likelier to fail, I'd like to use the latter as my in-use hard drive and the older one as my backup drive.  
This means I'm not doing a full backup and then keep both disks in sync, I'd like to clone the HDD and then use the newer HDD in daily life while the older HDD is only used as a backup disk which is kept in sync with the one I use (I hope that makes sense).

Because I'd like to clone the hard drive, at the time of the first backup they should be identical and basically no data should be backed up. The drive is encrypted with VeraCrypt but that shouldn't be a problem.

* I'd like to keep both drives in sync, how is this done with Vorta? If Vorta is not the right tool for that, please name an alternative tool and add a link if people already asked about it here.
* How can I make the older clone-source hard drive the full-disk backup location in Vorta? Is it enough if after cloning the disk I move all its contents into a directory that I set as the "Repository" in Vorta or how does that work?

----

I think this could be a very common use-case or it would be a common use-case if many people did full backups.  
Hence this could become a feature request, even if it only means changing the documentation. Here's why I think that doing this makes a lot of sense for most macOS/Linux users: backups should be run often and be incremental or sync (Vorta?), have a GUI and be relatively easy to use rather than requiring long tinkering or the command-line (Vorta), be done with a tool that is FOSS (Vorta), and be transferred to a storage location that is likely to fail first rather than the other way around (or not?).

It would be best if Vorta could do _all_ types of backup tasks - doing backups at all is already too difficult for most people so having to use a handful of different tools each for only some specific kind of sync or backup task just unnecessarily overcomplicates things when things could get unified into one best backup tool.

----

Asked about it here now: [How to keep two local hard drives in sync on Linux?](https://superuser.com/questions/1823915/how-to-keep-two-local-hard-drives-in-sync-on-linux)

**Comment by real-yfprojects** [comment]
I think rsync might be what you are looking for. Vorta is a GUI wrapper for the mighty [borg](https://github.com/borgbackup/borg). It is open for features supporting the use of borg but not for implementing new backup tools.

> backups should be run often and be incremental or sync (Vorta?)

There are big differences between your use case and the one borg/Vorta aim to support. You want to have two copies of your current file system state with one serving as backup storage. What you want is a file system sync tool. When we talk about a backup tool, we think of a tool that allows us to regularly make a backup by saving a snapshot of the current file system state which is then compressed and deduplicated with the existing snapshots. We store many snapshot of our data so that we can roll back to earlier versions. 

> It would be best if Vorta could do _all_ types of backup tasks - doing backups at all is already too difficult for most people so having to use a handful of different tools each for only some specific kind of sync or backup task just unnecessarily overcomplicates things when things could get unified into one best backup tool.

This currently falls out of the scope of the project. We develop this voluntarily so we have to limit the features we have to maintain.

BTW do full backups with Vorta requires it to be run as root which isn't supported at the moment. This is tracked in #1482.

**Reply by mYnDstrEAm** [reply]
So are you saying that would first need to be supported by borg?

I know that is the most common use-case of Vorta but that just comes back to wrote in the last two paragraphs...making things simple and not scattered across dozens of complicated tools each for a particular use-case even when that use-case is very useful and should be quite prevalent. In this case Vorta wouldn't compress the backup, it could also store multiple versions of changed files - the important thing is just that the entire drive is synced.

Ah right, forgot Vorta currently isn't capable of doing root backups anyway - I had that on the list and think I subscribed to an earlier issue about that, please implement that, I don't know why this doesn't seem to be of high-priority given how useful this would be and I can't stress enough how important that feature is.

Do you know of a tool or method that is better suited for my intended task - is the currently unmaintained BackInTime the best Linux tool for that?

**Reply by sten0** [reply]
@mYnDstrEAm, @real-yfprojects is saying that:

1. Borg is a tool that takes a list filesystem tree of /some/path /some/other/path, and creates an backup at /hard_drive-A/repo::some_archive_name
2. Vorta is a frontend for Borg which makes it easier to do this
3. You're requesting a A-to-B filesystem tree synchroniser.  Ie `rsync --delete /hard_drive-A/ /hard_drive-B/`
4. For an rsync GUI, use Grsync or RsyncOSX (as appropriate)
5. To tie them together, either write a custom launcher, or find someone to write one for you

If you want /rootfs level backups (rather than user-initiated and user-controlled) backups, then I recommend borgmatic.  It looks like Borgmatic can also create backups at /hard_drive-A/repo::some_archive_name as well as /hard_drive-B/repo::some_archive_name, as one operation, but I'm not sure if this is serialised or parallel.  If Borgmatic is impossible to learn to use, then you'll need to find (and possibly pay) someone to configure it for you, or to write a GUI for it.  To be fair, I also wish that Vorta could do everything that Borgmatic does :)

**Reply by real-yfprojects** [reply]
> To be fair, I also wish that Vorta could do everything that Borgmatic does

I do too! (:

**Reply by m3nu** [reply]
For CLI apps it's easier to add many features, because an extra flag or config file option doesn't take up space, except in man pages. For GUI apps, you'd end up with something like this: 😬

![wgetgui-screenshot](https://github.com/borgbase/vorta/assets/3916435/1f6f168a-5e49-4101-89ec-958fa1d45d9b)

**Reply by real-yfprojects** [reply]
Very true, that's why we have to limit ourselves.

**Reply by mYnDstrEAm** [reply]
@m3nu Not true at all but that would still be easier to use for many than CLI which could still be used as an alternative to the GUI. It's not true because one could make a step-by-step wizard GUI with few options per page and well-chosen default settings.

Thanks, @sten0 so probably I'll use Grsync. I hope this gets to be a feature in Borg and eventually Vorta. Currently I need to use 3-4 different backup apps: Vorta for most, BackInTime for backups that include root permission files ([please make that possible in Vorta](https://github.com/borgbase/vorta/discussions/801)), and now also  Grsync. I also use the Timeshift system restore utility (great GUI) and don't know if I should replace my use of BiT with it. I don't have an exotic use-case, I just don't want to lose my files or upload them all to a cloud and think it was good if the average person could just use a simple GUI for all common backup tasks.

I'd be interested in how Grsync [compares to BackInTime or Timeshift](https://github.com/borgbase/vorta/discussions/825), maybe I could just BiT/Timeshift. It should mirror all files but the bigger issue is **how the backup will proceed after the drive is mirrored**, I'd like to do incremental backups in the sense that the two drives should be in sync. I don't know how to best do that and probably need to figure out before doing fullbackup...obviously I don't want to do a full-backup then use the other hard drive then delete all the files on the second and do another full-backup, it should **sync** somehow. If I'm not mistaken yet another GUI tool would be needed ~~and [Unison](https://github.com/bcpierce00/unison) or [Syncthing](https://github.com/syncthing/syncthing) is probably best~~.

**Reply by real-yfprojects** [reply]
> it was good if the average person could just use a simple GUI for all common backup tasks.

Of course this would be nice. However there are a lot of different tools and setups people like to have such a GUI for. Therefore most people will benefit more from a more modularized approach where they can choose to combine different software to specifically suit there needs. Even more grave is the lack of resources. The Vorta project has very limited development resources which are already bound to supporting the very core goal of providing a GUI for borg for standard use cases. Therefore we don't really have a choice and cannot also provide a GUI for rsync.
